src/Repository.ts

```diff
@@ -357,54 +382,65 @@ export class MongoRepository<DOC, DTO = DOC> {
   private getCollection(): Promise<Collection<DOC>> {
     return new Promise<Collection<DOC>>(async (resolve, reject) => {
       const db = await this.dbSource.db;
-      db.collection(this.options.name, { strict: true }, async (err, collection) => {
-        let ourCollection = collection;
-        if (err) {
-          try {
-            ourCollection = await db.createCollection(this.options.name, {
-              size: this.options.size,
-              capped: this.options.capped,
-              max: this.options.max
-            });
-          } catch (createErr) {
-            if (createErr.codeName === 'NamespaceExists') {
-              // race condition. ignore for now, as I can't seem to get
-              // transactions to work in mongo 4.4 as yet
-              ourCollection = db.collection(this.options.name);
-            } else {
-              reject(err);
-            }
-          }
-        }
-
-        // assert indexes
-        if (this.options.indexes) {
-          for (const indexDefinition of this.options.indexes) {
-            try {
-              await ourCollection.createIndex(indexDefinition.fields, indexDefinition.options);
-            } catch (indexErr) {
-              if (
-                indexDefinition.overwrite &&
-                indexDefinition.options.name &&
-                indexErr.name === 'MongoError' &&
-                (indexErr.codeName === 'IndexKeySpecsConflict' || indexErr.codeName === 'IndexOptionsConflict')
-              ) {
-                // drop index and recreate
-                try {
-                  await ourCollection.dropIndex(indexDefinition.options.name);
-                  await ourCollection.createIndex(indexDefinition.fields, indexDefinition.options);
-                } catch (recreateErr) {
-                  reject(recreateErr);
-                }
-              } else {
-                reject(indexErr);
-              }
-            }
-          }
-        }
-
-        resolve(ourCollection);
-      });
+      let ourCollection;
+      const exist = (await db.listCollections({ name: this.options.name }).toArray())[0];
```

Comment:
I don't think we are really gaining anything with checking for existence, when we still can have a race condition in the creation of the collection in a distributed deployment.